app/controllers/tasks_controller.rb

```diff
@@ -50,19 +60,28 @@ def update
     end
   end
   def destroy
-    @task = Task.find(params[:id]).destroy
+    @task.destroy
     flash[:notice] = "タスクを削除しました！"
     redirect_to tasks_path
   end
 
   private
+  def set_task
+    @task = Task.find(params[:id])
```

Comment:
自分以外の人が作成したタスクも編集・削除できてしまいそうですが、下記のように制限しなくても大丈夫ですか？
```suggestion
    @task = current_user.tasks.find(params[:id])
```